indico/modules/events/api.py

```diff
@@ -255,8 +273,243 @@ def _iterate_objs(objIds):
+        }
+
+    fossils_mapping = {
+        'event': {
+            None: 'conferenceMetadata',
+            'contributions': 'conferenceMetadataWithContribs',
+            'subcontributions': 'conferenceMetadataWithSubContribs',
+            'sessions': 'conferenceMetadataWithSessions'
+        },
+        'contribution': {
+            'contributions': 'contributionMetadata',
+            'subcontributions': 'contributionMetadataWithSubContribs',
+            'sessions': 'contributionMetadataWithSubContribs'
+        },
+        'person': {
+            'Avatar': 'conferenceChairMetadata',
+            'ConferenceChair': 'conferenceChairMetadata',
+            'ContributionParticipation': 'contributionParticipationMetadata',
+            'SubContribParticipation': 'contributionParticipationMetadata'
+        }
+    }
+
+    def _build_event_basic_api_data(self, event):
+        return {
+            '_type': 'Conference',
+            'id': event.id,
+            'title': event.title,
+            'description': event.description,
+            'startDate': self._serialize_date(event.start_dt),
+            'timezone': event.timezone,
+            'endDate': self._serialize_date(event.end_dt),
+            'room': event.room_name,
+            'location': event.venue_name,
+            'address': event.address,
+            'type': event.type
+        }
+
+    def _build_event_api_data(self, event):
+        self._detail_level = get_query_parameter(request.args.to_dict(), ['d', 'detail'])
+        data = self._build_event_basic_api_data(event)
+        data.update({
+            '_fossil': self.fossils_mapping['event'].get(self._detail_level, None),
+            'categoryId': event.category_id,
+            'category': event.category.getTitle(),
+            'note': build_note_api_data(event.note),
+            'roomFullname': event.room_name,
+            'url': url_for('event.conferenceDisplay', confId=event.id, _external=True),
+            'modificationDate': self._serialize_date(event.as_legacy.getModificationDate()),
+            'creationDate': self._serialize_date(event.as_legacy.getCreationDate()),
+            'creator': self._serialize_person(event.creator, person_type='Avatar'),
+            'hasAnyProtection': event.as_legacy.hasAnyProtection(),
+            'roomMapURL': event.room.map_url if event.room else None,
+            'visibility': Conversion.visibility(event.as_legacy),
+            'folders': build_folders_api_data(event),
+            'chairs': self._serialize_persons(event.person_links, person_type='ConferenceChair'),
+            'material': build_material_legacy_api_data(event) + [build_note_legacy_api_data(event.note)]
+        })
+        if self._detail_level in {'contributions', 'subcontributions'}:
+            data['contributions'] = []
+            for contribution in event.contributions:
+                include_subcontrs = self._detail_level in {'subcontributions', 'sessions'}
```

Comment:
`include_subcontribs`.. let's not start using two different ways of abbreviating "contribution" ;)